components/bitcoin-topics/markdown-ui/stack-simulator.tsx

```diff
@@ -0,0 +1,214 @@
+        }
+    },
+    {
+        name: "OP_DUP",
+        execute: (stack) => {
+            if (stack.length < 1) return [...stack, "Error: Insufficient items"]
+            const a = stack[stack.length - 1]
+            return [...stack, a]
+        }
+    },
+    {
+        name: "OP_SWAP",
+        execute: (stack) => {
+            if (stack.length < 2) return [...stack, "Error: Insufficient items"]
+            const [b, a] = [stack.pop(), stack.pop()] as [StackItem, StackItem]
+            return [...stack, b, a]
+        }
+    }
+]
+
+interface StackSimulatorProps {
+    showOperations: boolean
+}
+
+const StackSimulator = ({ showOperations }: StackSimulatorProps) => {
+    const [stack, setStack] = useState<StackItem[]>([])
+    const [input, setInput] = useState<string>("")
+    const [lastOperation, setLastOperation] = useState<string | null>(null)
+
+    const push = (): void => {
+        if (input.trim()) {
+            const newItem = input.trim()
+            let stackItem: StackItem
+            if (newItem === "true" || newItem === "false") {
+                stackItem = newItem === "true"
+            } else if (!isNaN(Number(newItem))) {
+                stackItem = BigInt(newItem)
+            } else {
+                stackItem = newItem
+            }
+            setStack((prevStack) => [...prevStack, stackItem])
+            setInput("")
+            setLastOperation("push")
+        }
+    }
+
+    const pop = (): void => {
+        if (stack.length > 0) {
+            setStack((prevStack) => prevStack.slice(0, -1))
+            setLastOperation("pop")
+        }
+    }
+
+    const executeOpcode = (opcode: Opcode): void => {
+        const newStack = opcode.execute([...stack])
+        setStack(newStack)
+        setLastOperation(opcode.name)
+    }
+
+    const getItemColor = (item: StackItem): string => {
+        if (typeof item === "boolean") {
+            return item ? "bg-green-500 text-white" : "bg-red-500 text-white"
+        }
+        if (typeof item === "string" && item.startsWith("Error:")) {
+            return "bg-red-500 text-white"
+        }
+        return "bg-[#5A6270] text-[#E5E6F1]"
+    }
+
+    return (
+        <div className="w-full max-w-2xl mx-auto py-1">
+            <motion.div
+                className="bg-white dark:bg-[#272E35] p-4 sm:p-6 rounded-lg text-sm sm:text-base shadow-md"
+                initial={{ opacity: 0, y: 20 }}
+                animate={{ opacity: 1, y: 0 }}
+                transition={{ duration: 0.5 }}
+            >
+                <div className="text-xl sm:text-2xl font-bold mb-4 text-gray-800 dark:text-[#E5E6F1]">
+                    Bitcoin StackSimulator
+                </div>
+                <div className="flex space-x-2 mb-4">
+                    <input
+                        type="text"
+                        value={input}
+                        onChange={(e) => setInput(e.target.value)}
```

Comment:
Is it necessary to add the e.preventDefault() as this already work as is 🤔?

Reply:
The onChange event handler for the input does not trigger a form submission or any default action that needs to be prevented. I think for the current input handling, it is not required